tests/Resources/Page/ODH/JupyterHub/JupyterHubSpawner.robot

```diff
@@ -222,3 +223,9 @@ Login Via Button
    ...  Keyword.
    Click Element  xpath://a[@id='login']
    Wait Until Page Contains  Log in with
+
+Maybe Handle Server Not Running Page
+  ${SNR_visible} =  Run Keyword and Return Status   Wait Until Page Contains    Server not running  timeout=15
+  IF  ${SNR_visible}==True
+         Handle Server Not Running
+  END
```

Comment:
We already have the Keywords `Server Not Running Is Visible` and `Handle Server Not Running`, why define a new keyword?
If you really need a new keyword, you should use `Server Not Running Is Visible` in it to check whether you are in that page or not IMHO; that way if the text changes or we want to modify the timeout we only have to modify one keyword instead of two.

Reply:
I create another one because the original one use "Page Should Contains" but in this case the "Wait Until Page Contains" is needed, because it takes some time before appearing. 
As alternative, I could use the original one and adding a Sleep of 15sec before running the keyword. When the time is kind of variable, I prefer using the Wait until. Maybe it is something that could be changed in the original one.

Reply:
Then my suggestion would be to update `Server Not Running Is Visible` to use `Wait Until Page Contains`, with the timeout as an input variable. Then you can either reuse that keyword from within this one, or directly use a combination of `Server Not Running Is Visible` + `Handle Server Not Running` to achieve the same result.

Reply:
(Just keep in mind that if we are unsure about which page is being displayed, in the worst case scenario `Wait Until Page Contains` will "waste" `timeout` seconds waiting for something to appear.)

Reply:
I thought about changing the original one, but given that this PR would have been a "quick" one to fix the Anaconda TC in the smoke, I didn't want to change the already present. Same reason why I didn't include the "Server not running" keyword directly in the Fix Spawner Status.

But hey, I will do both the changes now to have a better code. Thanks

Reply:
done in commit https://github.com/red-hat-data-services/ods-ci/pull/145/commits/bb84fdd4d769dd8d59087076e85e69c6f329d9c2

Implemented in a way to avoid the waste of timeout at the beginning of "Fix Spawner Status" keyword.

Reply:
> Implemented in a way to avoid the waste of timeout at the beginning of "Fix Spawner Status" keyword.
I have to change this again because it appears to have a situation where it fails. This situation didn't show up in the Friday testing, but it did today. I'm working on it